internal/audit/interceptor.go

```diff
@@ -47,12 +54,14 @@ func NewUnaryInterceptor(log Log, exclude ExcludeMethod) grpc.UnaryServerInterce
 				StatusCode: uint32(status.Code(err)),
 			}
 
-			md, ok := metadata.FromIncomingContext(ctx)
-			if ok {
-				entry.Metadata = make(map[string]*auditv1.MetaValues, len(md))
-				for key, values := range md {
-					if _, ok := excludeMetadataKeys[key]; !ok {
-						entry.Metadata[key] = &auditv1.MetaValues{Values: values}
+			if !(len(conf.ExcludeMetadataKeys) == 0 && len(conf.IncludeMetadataKeys) == 0) {
```

Comment:
Let's do this check once and save the result to a variable so that it doesn't need to be done on every request.